cdap-storage-spi/src/main/java/io/cdap/cdap/spi/data/StructuredTable.java

```diff
@@ -194,6 +194,27 @@ default CloseableIterator<StructuredRow> scan(Range keyRange, int limit,
     throw new UnsupportedOperationException("No supported implementation.");
   }
 
+  /**
+   * Read a set of rows from the table matching the index and the key range. The rows returned will
+   * be sorted on given indexed field.
+   *
+   * @param keyRange key range for the scan
+   * @param limit maximum number of rows to return
+   * @param filterIndexes the index to filter upon
+   * @param orderByField the field to sort upon
+   * @param sortOrder defined primary key sort order. Note that the comparator used is specific
+   *     to the underlying store and is not necessarily lexicographic.
+   * @return a {@link CloseableIterator} of rows
+   * @throws InvalidFieldException if the field is not part of the table schema, or is not an
+   *     indexed column, or the type does not match the schema
+   * @throws IOException if there is an error scanning the table
+   */
+  default CloseableIterator<StructuredRow> scan(Range keyRange, int limit,
```

Comment:
Currently only implemented in PostgresSql. Will handle nosql in a seperate PR

Reply:
this isn't needed anymore right?

Reply:
This is needed as no current functions support all of the options of filterindexes, orderedbyField and sortorder